dbms/tests/queries/0_stateless/01012_empty_in_inner_right_join.sql

```diff
@@ -0,0 +1,2 @@
+SELECT count() FROM system.numbers WHERE number IN (SELECT toUInt64(1) WHERE 0);
```

Comment:
Please also add a test case with RIGHT JOIN and LEFT JOIN.
And a case with NOT IN empty set.
And a case where IN set is not in WHERE clause: `SELECT avg(x IN (empty set)) FROM ...`
And a case when you have multiple sets, some of them are in WHERE and some of them are not.

Reply:
And add a case like `WHERE NOT ignore(x IN empty set)`